### PR TITLE
Model config dict constructor

### LINKED ISSUE #593: Python constructor for model_config
Create python constructor and equal operator for model_config

### PR DESCRIPTION
**Pull request by ManInFez**
New constructor and test data for ModelConfig 

Resolves #593

### CODE REVIEW

File: python/res/enkf/config_keys.py [hunk 1]
```diff
@@ -148,17 +155,32 @@ class ConfigKeys:
...
     RUNPATH_FILE     = _runpath_file()
     RUNPATH_LIST_FILE = ".ert_runpath_list"
+    GEN_DATA         = _gen_data()
+    RESULT_FILE      = _result_file()
+    REPORT_STEPS     = _report_steps()
+    INPUT_FORMAT     = _input_format()
+    ECL_FILE         = _ecl_file()
+    OUTPUT_FORMAT    = _output_format()
+    INIT_FILES       = _init_files()
+    LICENSE_PATH     = _license_path_key()
```
Comment (markusdregi) [inline python/res/enkf/config_keys.py:173]: What a beautiful roller coaster of white space alignment.

But, it is not your fault that white space alignment is inherently broken! Blame it on the fool that initiated it in this file instead (in other words, a younger me)...

File: python/res/enkf/model_config.py [hunk 2]
```diff
@@ -16,17 +16,35 @@
...
+from res.enkf.util import TimeMap
+
+import os
 
 
 class ModelConfig(BaseCClass):
     TYPE_NAME = "model_config"
 
     _alloc                       = ResPrototype("void*  model_config_alloc( config_content, char*, ext_joblist, int, ecl_sum)", bind=False)
+    _alloc_full = ResPrototype("void*  model_config_alloc_full( int, \
```
Comment (markusdregi) [inline python/res/enkf/model_config.py]: Yet another interesting white space indentation, but this one there might be hope for...

File: python/res/enkf/model_config.py [hunk 3]
```diff
@@ -48,12 +66,107 @@ class ModelConfig(BaseCClass):
     _get_obs_config_file         = ResPrototype("char* model_config_get_obs_config_file(model_config)")
     _get_data_root               = ResPrototype("char* model_config_get_data_root(model_config)")
     _set_data_root               = ResPrototype("void model_config_get_data_root(model_config, char*)")
+    _get_time_map = ResPrototype("void model_config_get_external_time_map(model_config)")
+
+    def __init__(self, data_root, joblist, last_history_restart, refcase, config_content=None, config_dict=None, is_reference=False):
```
Comment (markusdregi) [inline python/res/enkf/model_config.py]: Please break over multiple lines

File: python/res/enkf/model_config.py [hunk 4]
```diff
@@ -48,12 +66,107 @@ class ModelConfig(BaseCClass):
...
+                ens_path = os.path.realpath(ens_path)
+
+            # JOBNAME_KEY
+            job_name = config_dict.get(ConfigKeys.JOBNAME)
+
+            # FORWARD_MODEL_KEY
+            forward_model = ForwardModel(ext_joblist=joblist)
+            for job_description in config_dict.get(ConfigKeys.SIMULATION_JOB, []):
+                job = forward_model.add_job(job_description['NAME'])
+                args = StringList()
```
Comment (markusdregi) [inline python/res/enkf/model_config.py]:
```
args = StringList(job_description['ARGUMENTS'])
```
?

File: python/res/enkf/model_config.py [hunk 5]
```diff
@@ -137,5 +248,50 @@ def num_realizations(self):
...
+        if self.get_max_internal_submit() != other.get_max_internal_submit():
+            return False
+
+        if self.getGenKWExportName() != other.getGenKWExportName():
+            return False
+
+        if self.get_time_map() != other.get_time_map():
+            return False
+
+        if self.getForwardModel() != other.getForwardModel():
```
Comment (markusdregi) [inline python/res/enkf/model_config.py:295]: Is this not a bit optimistic? :)

**Review by markusdregi — COMMENTED** [review]
All in all, this looks very good :+1:

File: python/res/enkf/model_config.py (hunk 5 above)
Comment (ManInFez) [reply python/res/enkf/model_config.py:295]: Isn't it better than to be pessimistic?

Comment (markusdregi) [reply python/res/enkf/model_config.py:295]: Never mind!

**Review by markusdregi — APPROVED** [review]

**Pull request merged**